### PR TITLE
Rename namespace `vec256` to `vec`, struct `Vec256` to `Vectorized` (and other related classes/structs)

### PR DESCRIPTION
**Pull request by imaginary-person**
### Summary
In order to make it more convenient for maintainers to review the ATen AVX512 implementation, the namespace `vec256` is being renamed to `vec` in this PR, as modifying 77 files & creating 2 new files only took a few minutes, as these changes aren't significant, so fewer files would've to be reviewed while reviewing #56992.
The struct `Vec256` is not being renamed to `Vec`, but `Vectorized` instead, because there are some `using Vec=` statements in the codebase, so renaming it to `Vectorized` was more convenient. However, I can still rename it to `Vec`, if required.

### Changes made in this PR -
Created `aten/src/ATen/cpu/vec` with subdirectory `vec256` (vec512 would be added via #56992).
The changes were made in this manner -

1. First, a script was run to rename `vec256` to `vec` & `Vec` to `Vectorized` -
```
# Ref: https://stackoverflow.com/a/20721292
cd aten/src
grep -rli 'vec256\/vec256\.h' * | xargs -i@ sed -i 's/vec256\/vec256\.h/vec\/vec\.h/g' @
grep -rli 'vec256\/functional\.h' * | xargs -i@ sed -i 's/vec256\/functional\.h/vec\/functional\.h/g' @
grep -rli 'vec256\/intrinsics\.h' * | xargs -i@ sed -i 's/vec256\/intrinsics\.h/vec\/vec256\/intrinsics\.h/g' @
grep -rli 'namespace vec256' * | xargs -i@ sed -i 's/namespace vec256/namespace vec/g' @
grep -rli 'Vec256' * | xargs -i@ sed -i 's/Vec256/Vectorized/g' @
grep -rli 'vec256\:\:' * | xargs -i@ sed -i 's/vec256\:\:/vec\:\:/g' @
grep -rli 'at\:\:vec256' * | xargs -i@ sed -i 's/at\:\:vec256/at\:\:vec/g' @
cd ATen/cpu
mkdir vec
mv vec256 vec
cd vec/vec256
grep -rli 'cpu\/vec256\/' * | xargs -i@ sed -i 's/cpu\/vec256\//cpu\/vec\/vec256\//g' @
grep -rli 'vec\/vec\.h' * | xargs -i@ sed -i 's/vec\/vec\.h/vec\/vec256\.h/g' @
```

2. `vec256` & `VEC256` were replaced with `vec` & `VEC` respectively in 4 CMake files.

3. In `pytorch_vec/aten/src/ATen/test/`, `vec256_test_all_types.h` & `vec256_test_all_types.cpp` were renamed.

4. `pytorch_vec/aten/src/ATen/cpu/vec/vec.h` & `pytorch_vec/aten/src/ATen/cpu/vec/functional.h` were created.
Both currently have one line each & would have 5 when AVX512 support would be added for ATen.

### CODE REVIEW

**Comment by imaginary-person** [comment]
Hello @ezyang, @malfet, @VitalyFedyunin & @ngimel,

This PR would make reviewing #56992 more convenient for you, as fewer files would then have to be reviewed.
BTW, #56992 is almost ready for review, except for Windows builds (It'd be great if @skyline75489 or @peterjc123 could help with it).

Also, CI checks (tests, not builds) currently don't seem to use executors with AVX512 support. So, please confirm if executors for some CI checks would've to be changed after adding AVX512 support, or if I should temporarily change executors of some CI checks for that PR, so that AVX512 kernels can be tested on CI. Thank you!

**Comment by ezyang** [comment]
thanks, splitting the PRs like this is highly appreciated

File: aten/src/ATen/cpu/vec/vec256/intrinsics.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
         (defined(__VEC__) || defined(__ALTIVEC__))
 /* XLC or GCC-compatible compiler, targeting PowerPC with VMX/VSX */
 #include <altivec.h>
-/* We need to undef those tokens defined by <altivec.h> to avoid conflicts
+/* We need to undef those tokens defined by <altivec256.h> to avoid conflicts
```
Comment (ezyang) [inline aten/src/ATen/cpu/vec/vec256/intrinsics.h:35]: whoops!

**Comment by ezyang** [comment]
I need CI to run on this PR, it seems like it is being skipped right now

**Comment by ezyang** [comment]
btw, pro-tip for the future, try to do file rename separately from codemod, it's easier for VCS systems to deal with it.

**Comment by imaginary-person** [comment]
Sorry about that, @ezyang! I triggered CI again by modifying a comment. I apologize for the inconvenience, but you might've to re-import the PR.

**Review by ezyang — APPROVED** [review]

**Review by VitalyFedyunin — APPROVED** [review]

**Comment by ezyang** [comment]
warning needs fbcode changes, due to downstream users of vec256

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 2]
```diff
@@ -434,7 +434,7 @@ void qrelu6_kernel(const Tensor& qx, Tensor& qy) {
         qx.q_scale(),
         qx.q_zero_point(),
         c10::nullopt);
-    using Vec = Vec256<scalar_t>;
+    using Vec = Vectorized<scalar_t>;
```
Comment (peterjc123) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:437]: Windows CI claims that functions like `relu`, `relu6` are missing in the namespace `Vectorized<scalar_t>`. I wonder if the function implementations for some types are missing. Would you please expand `scalar_t` into all possible types defined in `AT_DISPATCH_QINT_TYPES` at a place for testing?

**Pull request closed without merging**